src/skin/legacyskinparser.cpp

```diff
@@ -1084,6 +1089,30 @@ QWidget* LegacySkinParser::parseCoverArt(QDomElement node) {
     return pCoverArt;
 }
 
+void LegacySkinParser::parseSingletonDefinition(QDomElement node) {
+    // The actual singleton definition is identical to any other WWidgetGroup.
+    QWidget* child_group = parseWidgetGroup(node);
+    m_pContext->getSingletonMap()->defineSingleton(child_group->objectName(),
+                                                   child_group);
+    child_group->hide();
+}
+
+QWidget* LegacySkinParser::parseSingletonContainer(QDomElement node) {
```

Comment:
So I made this change in a separate branch as an experiment:  https://github.com/ywwg/mixxx/pull/4/files

I see what you mean about making the parsing standard, but there are a few things I don't like:

* SingletonContainer is now tightly coupled with SkinContext, with a weird cyclic dependency that means they can't be separated
* It's possible to have a badly-constructed container with null layout and widget, since that information can't be constructed until setup() time.

But I don't know how much of that is this specific widget, and how much is the skin design in general.  I guess my last-ditch argument is "well this isn't a standard widget ¯\_(ツ)_/¯" but that's pretty lame.  What do you think?